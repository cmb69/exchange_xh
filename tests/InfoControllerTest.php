<?php

namespace Exchange;

use ApprovalTests\Approvals;
use PHPUnit\Framework\TestCase;
use Plib\FakeSystemChecker;
use Plib\View;

class InfoControllerTest extends TestCase
{
    /** @var array<string,string> */
    private $lang;

    /** @var SystemChecker */
    private $systemChecker;

    /** @var View */
    private $view;

    public function setUp(): void
    {
        $this->systemChecker = new FakeSystemChecker();
        $this->lang = XH_includeVar("./languages/en.php", "plugin_tx")["exchange"];
        $this->view = new View("./views/", $this->lang);
    }

    private function sut(): InfoController
    {
        return new InfoController(
            "./plugins/exchange/",
            $this->systemChecker,
            $this->view
        );
    }

    public function testShowsPluginInfo(): void
    {
        $response = $this->sut()();
        Approvals::verifyHtml($response);
    }
}
