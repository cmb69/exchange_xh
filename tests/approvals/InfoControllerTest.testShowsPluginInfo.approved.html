<h1>Exchange</h1>
<img src="./plugins/exchange/exchange.png" class="exchange_logo" alt="Spinning Arrows">
<p>Version: 1.1-dev</p>
<p>
    Copyright 2017 <a href="http://3-magi.net/">Christoph M. Becker</a>
</p>
<p class="exchange_license">
    Exchange_XH is free software: you can redistribute it and/or modify it
    under the terms of the GNU General Public License as published by the Free
    Software Foundation, either version 3 of the License, or (at your option)
    any later version.
</p>
<p class="exchange_license">
    Exchange_XH is distributed in the hope that it will be useful, but
    <em>without any warranty</em>; without even the implied warranty of
    <em>merchantability</em> or <em>fitness for a particular purpose</em>. See
    the GNU General Public License for more details.
</p>
<p class="exchange_license">
    You should have received a copy of the GNU General Public License along with
    Exchange_XH. If not, see <a
    href="http://www.gnu.org/licenses/">http://www.gnu.org/licenses/</a>.
</p>
<div class="exchange_syscheck">
    <h2>System check</h2>
    <p class="xh_fail">Checking that PHP version ≥ 7.1.0 … failure</p>
    <p class="xh_fail">Checking that the PHP extension 'DOM' is loaded … failure</p>
    <p class="xh_fail">Checking that the PHP extension 'SimpleXML' is loaded … failure</p>
    <p class="xh_fail">Checking that CMSimple_XH version ≥ 1.7.0 … failure</p>
    <p class="xh_warning">Checking that './plugins/exchange/css/' is writable … warning</p>
    <p class="xh_warning">Checking that './plugins/exchange/languages/' is writable … warning</p>
</div>
